src/main/java/org/spongepowered/common/item/recipe/smelting/SpongeSmeltingRecipeBuilder.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.item.recipe.smelting;
+
+import static com.google.common.base.Preconditions.*;
+
+import org.spongepowered.api.item.inventory.ItemStackSnapshot;
+import org.spongepowered.api.item.recipe.smelting.SmeltingRecipe;
+
+import java.util.function.Predicate;
+
+import javax.annotation.Nonnull;
+
+public class SpongeSmeltingRecipeBuilder implements SmeltingRecipe.Builder {
+    private ItemStackSnapshot exemplaryResult;
+    private ItemStackSnapshot exemplaryIngredient;
+    private Predicate<ItemStackSnapshot> ingredientPredicate;
+    private Double experience;
```

Comment:
https://github.com/SpongePowered/SpongeCommon/pull/1119/files#diff-623d28dbc3a9aefb69d7573e328abae8R118

Going to need to change this then